src/wp-includes/html-api/class-wp-xml-tag-processor.php

```diff
@@ -402,6 +402,22 @@ class WP_XML_Tag_Processor {
 	 */
 	protected $parser_state = self::STATE_READY;
 
+	/**
+	 * Whether we stopped at an incomplete text node.
+	 *
+	 * If we are before the last tag in the document, every text
+	 * node is incomplete until we find the next tag. However,
+	 * if we are after the last tag, an incomplete all-whitespace
+	 * node may either mean we're the end of the document or
+	 * that we're still waiting for more data/
+	 *
+	 * This flag allows us to differentiate between these two
+	 * cases in context-aware APIs such as WP_XML_Processor.
+	 *
+	 * @var bool
+	 */
+	protected $is_incomplete_text_node = false;
```

Comment:
this name confused me. I think something like `$inside_document_trailing_whitespace` could be clearer, but I could still be confused